### PR TITLE
chore(gpu): rework single carry prop to avoid using local streams

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
Multi-GPU tests running here: https://github.com/zama-ai/tfhe-rs/actions/runs/12393147259/job/34594035478

**Comment by agnesLeroy** [comment]
On 1xH100 there does not seem to be any effect on the latency & throughput benchmark measurements.

File: backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h [hunk 1]
```diff
@@ -1929,39 +1923,57 @@ template <typename Torus> struct int_sc_prop_memory {
...
+          (Torus *)malloc((num_radix_blocks + 1) * sizeof(Torus));
+      for (int index = 0; index < num_radix_blocks + 1; index++) {
+        if (index < num_radix_blocks) {
+          h_lut_indexes[index] = 0;
+        } else {
+          h_lut_indexes[index] = 1;
+        }
+      }
+      cuda_memcpy_async_to_gpu(
+          lut_message_extract->get_lut_indexes(0, 0), h_lut_indexes,
```
Comment (guillermo-oyarzun) [inline backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:1969]: should this be ->get_lut_indexes(0,1) ?

**Review by guillermo-oyarzun — COMMENTED** [review]
This is change is a bit complex. Just to be extra safe, it should be verified with: add, unsigned_overflowing_add, signed_overflowing_add, unsigned_overflowing_scalar_add, just to test all combinations

File: backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h (hunk 1 above)
Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:1969]: No here I copy the full array so it's expected I get the lut_indexes of lut 0

**Comment by agnesLeroy** [comment]
All multi-GPU tests have passed, I also ran those functions locally they were passing.

**Comment by guillermo-oyarzun** [comment]
> All multi-GPU tests have passed, I also ran those functions locally they were passing.

ok, then good to go!

**Pull request merged**